atdgen/src/json.ml

```diff
@@ -33,7 +37,7 @@ type json_repr =
   | Option
   | Record of json_record
   | String
-  | Sum
+  | Sum of json_adapter option
```

Comment:
At this stage, only atd variant types support the `<json adapter=...>` annotation.

It may be useful to support it on records as well; see https://github.com/mjambon/atd/issues/57 for an example.